pkg/server/endpoints/endpoints.go

```diff
@@ -294,7 +296,21 @@ func (e *Endpoints) runTCPServer(ctx context.Context, server *grpc.Server) error
 		return err
 	case <-ctx.Done():
 		log.Info("Stopping Server APIs")
-		server.Stop()
+
+		stopComplete := make(chan struct{})
+		go func() {
+			log.Info("Attempting graceful stop")
+			server.GracefulStop()
+			close(stopComplete)
+		}()
+
+		shutdownDeadline := time.After(e.GracefulStopWait)
+		select {
+		case <-shutdownDeadline:
+			log.Infof("Graceful stop unsuccessful, forced stop after %s", e.GracefulStopWait.String())
```

Comment:
I think we should also call `server.Stop()` in this case since the `server.GracefulStop()` might be still be doing some work.

Reply:
ahh! I had done this and then removed it erroneously in a refactor :\